src/apps/backend/modules/account/types.ts

```diff
@@ -1,24 +1,43 @@
 import { ApplicationError } from '../application';
 import { HttpStatusCodes } from '../http';
 
+export type Nullable<T> = T | null;
+
+export class PhoneNumber {
+  countryCode: string;
+  phoneNumber: string;
+
+  constructor(countryCode: string, phoneNumber: string) {
+    this.countryCode = countryCode;
+    this.phoneNumber = phoneNumber;
+  }
+
+  toString(): string {
+    return `${this.countryCode} ${this.phoneNumber}`;
+  }
+}
+
 export class Account {
-  id: string;
   firstName: string;
+  hashedPassword: string;
+  id: string;
   lastName: string;
+  phoneNumber: PhoneNumber;
   username: string;
-  hashedPassword: string;
 }
 
 export type CreateAccountParams = {
   firstName: string;
   lastName: string;
   password: string;
+  phoneNumber?: PhoneNumber;
   username: string;
 };
 
 export type AccountSearchParams = {
-  password: string;
-  username: string;
+  password?: string;
+  phoneNumber?: PhoneNumber;
+  username?: string;
 };
```

Comment:
Refactor this as 

`type AccountSearchParams = AccountSearchByEmailParams | AccountSearchByPhoneParams`